src/ContMechTensors.jl

```diff
@@ -293,4 +294,54 @@ end
     end
 end
 
+"""
+```julia
+basevec(::Type{Vec{dim, T}})
+basevec(::Type{Vec{dim, T}}, i)
+basevec(::Vec{dim, T})
+basevec(::Vec{dim, T}, i)
+```
+Return a tuple with the base vectors corresponding to the dimension `dim` and type
+`T`. An optional integer `i` can be used to extract the i:th base vector.
+The alias `eᵢ` can also be used.
+
+**Example:**
+
+```jldoctest
+julia> eᵢ(Vec{2, Float64})
+([1.0,0.0],[0.0,1.0])
+
+julia> eᵢ(Vec{2, Float64}, 2)
+2-element ContMechTensors.Tensor{1,2,Float64,2}:
+ 0.0
+ 1.0
+```
+"""
+@generated function basevec{dim, T}(::Type{Vec{dim, T}})
```

Comment:
Maybe just write out the three functions here instead of using a generated function. Would also remove the need for the dim error checker.